bash_completion/kerl

```diff
@@ -7,89 +7,123 @@
 # shellcheck disable=SC2312  # Consider invoking this command separately to avoid masking its return value (or use '|| true' to ignore)
 
 _kerl() {
-    local cur prev
-    if type _get_comp_words_by_ref &>/dev/null; then
-        _get_comp_words_by_ref cur prev
-    else
-        cur=$2 prev=$3
-    fi
+    local command cur prev releases builds installations file
+
+    command=${COMP_WORDS[1]}
+    cur=${COMP_WORDS[${#COMP_WORDS[@]} - 1]}
+    prev=${COMP_WORDS[${#COMP_WORDS[@]} - 2]}
 
-    case $prev in
-    kerl)
+    if [ "$COMP_CWORD" -eq 1 ]; then
         COMPREPLY=($(compgen -W 'build install build-install deploy update list delete path active plt status prompt cleanup emit-activate upgrade version' -- "$cur"))
-        ;;
-    list)
-        COMPREPLY=($(compgen -W 'releases builds installations' -- "$cur"))
-        ;;
+    fi
+
+    case $command in
     build)
         if [ "$COMP_CWORD" -eq 2 ]; then
-            if [ -f "$HOME"/.kerl/otp_releases ]; then
-                RELEASES=$(cat "$HOME"/.kerl/otp_releases)
+            file="$HOME"/.kerl/otp_releases
+            if [ -f "$file" ]; then
+                releases=$(cat "$file")
```

Comment:
This structure `file ... if -f file ... COMPREPLY` was used in most places, while also restricting input with COMP_CWORD wherever required. The file thus feel more readable. And to top it of we only assign to `COMPREPLY` when we actually have stuff to assign to it.